src/sys/fuchsia/handles.rs

```diff
@@ -0,0 +1,66 @@
+use {io, poll, Evented, Ready, Poll, PollOpt, Token};
+use magenta::HandleBase;
+use std::sync::Mutex;
+
+/// Wrapper for registering a `HandleBase` type with mio.
+#[derive(Debug)]
+pub struct EventedHandle<T> where T: HandleBase {
```

Comment:
Oh yeah this seems great to expose, having the ability to do `PollEvented<CustomType>` is crucial for exposing platform goodies! (this is how `tokio-process` works at all). What I mostly meant was that this was exposed pretty differently than other platforms. On Unix we've got `EventedFd` which notably doesn't take ownership and also just works with a raw `&i32`. On Windows is... slightly different but basically works with `AsRawHandle` in the standard library.

That being said I don't mind having the design here basically be up to y'all. I had a brief chat with @cramertj on IRC about API stability here and there's not too many concerns. I think it'd be fine to just try to make the fields private here for now and worry about stability later.